### PR TITLE
Add dev documentation style guide.

### PR DESCRIPTION
**Pull request by MFraters**
part of #532.

### CODE REVIEW

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md [hunk 1]
```diff
@@ -2,6 +2,164 @@
 Documentation style guide
 =========================
 
-```{todo}
-Explain how to style the markdown files correctly
+Not only are there many ways to style the documents, there are also many ways with sphinx to reach the same output. To attempt to unify the style of the documentation and the documentation code, this page contains some general styling recommendations.
+
+# Page titles
+
+Page titles should always use equal signs below it (`=`) to create the heading. This creates a clear visual destincation between normal headers and the page title.
```
Comment (tjhei) [inline doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:9]:
```suggestion
Page titles should always use equal signs below it (`=`) to create the heading. This creates a clear visual distinction between normal headers and the page title.
```

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md [hunk 2]
```diff
@@ -2,6 +2,164 @@
...
+
+# Page label
+
+The page label should always be a path, sepearted by colons (`:`) in the following style: `part:dev_manual:chap:contribute_to_doc:sec:doc_style_guide`.
+
+# Including advanced elements
+
+When possible, the use of colons (`:`) should be preffered over the use of grave accents (\`) when creating advaced elements such as todo notes, figures code blocks or grids. This means that:
```
Comment (tjhei) [inline doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:17]:
```suggestion
When possible, the use of colons (`:`) should be preferred over the use of grave accents (\`) when creating advanced elements such as todo notes, figures, code blocks, or tables. This means that:
```

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md [hunk 3]
```diff
@@ -2,6 +2,164 @@
...
+
+::::{code-block} md
+:::{note}
+My Note
+:::
+::::
+
+is preffered over the use of
```
Comment (tjhei) [inline doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:25]:
```suggestion
is preferred over the use of 
```

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md [hunk 4]
```diff
@@ -2,6 +2,164 @@
...
+
+::::{code-block} md
+```{note}
+My Note
+````
+::::
+
+# Including World Builder files
+
+When including a world builder file, it is imporant to be able to focus on the parts which are important, while also enabling the reader to see the  surrounding context and the whole file. Furthermore, the user should have the capability to run the models themselfs and get the same output as shown. To standerize this the following structure is strongly encuraged.
```
Comment (tjhei) [inline doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:35]:
```suggestion
When including a world builder file, it is imporant to be able to focus on the parts which are important, while also enabling the reader to see the  surrounding context and the whole file. Furthermore, the user should have the capability to run the models themselves and get the same output as shown. To standardize this the following structure is strongly encouraged. 
```

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md [hunk 5]
```diff
@@ -2,6 +2,164 @@
...
+```{note}
+My Note
+````
+::::
+
+# Including World Builder files
+
+When including a world builder file, it is imporant to be able to focus on the parts which are important, while also enabling the reader to see the  surrounding context and the whole file. Furthermore, the user should have the capability to run the models themselfs and get the same output as shown. To standerize this the following structure is strongly encuraged. 
+
+1. The world builder file and a corresponding grid file need to be stored seperately, either in the cookbook, or in the _static directory.
```
Comment (tjhei) [inline doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:37]:
```suggestion
1. The world builder file and a corresponding grid file need to be stored separately, either in the cookbook, or in the _static directory.
```

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md [hunk 6]
```diff
@@ -2,6 +2,164 @@
...
+2. The use of a tab-set with a tab with important lines and a tab with a the full world builder file, highlighting imporant lines
```
Comment (tjhei) [inline doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:38]:
```suggestion
2. The use of a tab-set with a tab with important lines and a tab with the full world builder file, highlighting important lines.
```

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md [hunk 7]
```diff
@@ -2,6 +2,164 @@
...
+3. Adding at least a link to the world bulider file and a grid file to visualize it, preferably also a link to a `.pvtu` file which allows to get the same output as the figure, if present.
```
Comment (tjhei) [inline doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:39]:
```suggestion
3. Adding at least a link to the world builder file and a grid file to visualize it, preferably also a link to a `.pvtu` file which allows to get the same output as the figure, if present.
```
Do you mean vtu and pvsm here?

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md [hunk 8]
```diff
@@ -2,6 +2,164 @@
...
+
+::::::
+
+
+```{figure} ../../../../doc/sphinx/_static/images/user_manual/basic_starter_tutorial/BST_05.png
+:name: BST_05_example
+:alt: Basic Starter Tutorial section 5. 
+:align: center
+
+Image description
```
Comment (tjhei) [inline doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:164]: Add a newline at the end.

**Review by tjhei — APPROVED** [review]

File: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md (hunk 8 above)
Comment (MFraters) [reply doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:164]: between Image description and the ``` or at the end of the document?

Comment (tjhei) [reply doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_style_guide.md:164]: My diff view shows that there is no newline at the end of the file, which I think is neater to have.

**Pull request merged**